platform/admin/configurator/admin-configurator-api/src/main/java/org/codice/ddf/admin/configurator/Configurator.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+/**
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface Configurator {
+    /**
+     * Sequentially invokes all the transaction's operations, committing their changes. If a failure
+     * occurs during the processing, a rollback is attempted of those handlers that had already been
+     * committed.
+     * <p>
+     * In the case of a successful commit, changes should be logged.
+     *
+     * @param auditMessage In the case of a successful commit, the message to pass to a
```

Comment:
❓Should we mention audit messages and security loggers here or should we keep things more generic and simply use concepts like transaction description and parameters? If we were to have a `ConfigurationTransactionFactory`, those could actually be passed in when a `ConfigurationTransaction` is created and used internally however the implementation sees fit.